### PR TITLE
Fix SysOutputInfo.write_to_directory.

### PR DESCRIPTION
**Pull request by odashi**
This PR implements detailed behaviors of `SysOutputInfo.write_to_directory` in various situations of the target filesystems. Also adds `overwrite` option to avoid unintended changes on the existing files.

This PR also removes `test_output_dir` which causes some errors by applying this change. This config should not be used since running multiple tests on the shared environment brings causality problems.

Fixes https://github.com/neulab/ExplainaBoard/issues/382

### CODE REVIEW

**Comment by odashi** [comment]
@neubig ping

File: explainaboard/info.py [hunk 1]
```diff
@@ -109,13 +109,49 @@ def replace_nonstring_keys(self, data):
...
-            else os.path.join(dataset_info_dir, file_name)
+    def write_to_directory(
+        self,
+        dataset_info_dir: str,
+        file_name: str | None = None,
+        overwrite: bool = False,
+    ) -> None:
+        """Write `SysOutputInfo` as JSON to `dataset_info_dir`.
+
+        This function is not thread-safe. Modify the target directory/files by other
```
Comment (neubig) [inline explainaboard/info.py:120]:
```suggestion
        This function is not thread-safe. Modification of the target directory/files by other
```

File: explainaboard/info_test.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            """Helper to check if the file is written correctly."""
+            self.assertTrue(path.exists())
+            with path.open() as f:
+                self.assertEqual(f.readline(), "{\n")  # Checks only the first line.
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            td = pathlib.Path(tmpdir)
+
+            # w/o directory
+            dir1 = td / "dir1"
```
Comment (neubig) [inline explainaboard/info_test.py:25]: Just a comment: I didn't know this way of specifying paths, very nice 😄

**Review by neubig — APPROVED** [review]
LGTM!

**Pull request merged**